src/ViewRenderer/Interfaces/ThemeServiceInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\ViewRenderer\Interfaces;
+
+interface ThemeServiceInterface
+{
+    /**
+     * Gets the current theme.
+     * This correctly considers the portal or backend but can also be overwritten by setting the theme beforehand - which is then always returned.
+     *
+     * @return \TdbPkgCmsTheme|null
+     */
+    public function getTheme(?\TdbCmsPortal $portal): ?\TdbPkgCmsTheme;
+
+    public function setOverrideTheme(?\TdbPkgCmsTheme $themeOverride): void;
```

Comment:
I would remove this method from the interface (but keep it in the concrete implementation). This is a special case we need only for the console command that is used to generate all themes. We do not want to make that option available to other code.